client/src/screens/training/controller.js

```diff
@@ -0,0 +1,45 @@
+export const controller = (serviceLocator) => {
+  const { getStoreData } = serviceLocator.getStore();
+  const { navigationApi } = serviceLocator.getAPIs();
+
+  const getData = () => getStoreData(controller.storeDataAccessors);
+  const getParams = () => navigationApi.getPathParams(
+    navigationApi.routes.openTraining,
+  );
+  
+  return {
+    getTraining: () => {
+      const params = getParams();
+      const trainings = getData().trainings;
+      const training = trainings.find(training => training.id === params.training);
+      return training;
```

Comment:
1. check if training is not undefined
2. then `training.exercises.map` to add new property `setsPreview` to each exercise
3. return training with new exercises

Reply:
`setsPreview: getSets(exercise)`